### ISSUE TITLE
PHP Fatal error:  Uncaught TypeError: round(): Argument #1 ($num) must be of type int|float, string given in /usr/share/cacti/site/poller.php:333

### ISSUE THREAD

**Issue by ufehembach** [Cacti/cacti#5320]
# Describe the bug

OS: Raspbian GNU/Linux 11 (bullseye) aarc 
Host: Raspberry Pi 4 Model B Rev 1.5 
Kernel: 6.1.21-v8+ 
Uptime: 2 days, 2 hours, 21 mins 
Packages: 1620 (dpkg) 
Shell: bash 5.1.4 
Resolution: 1360x768 
Terminal: /dev/pts/2 
CPU: BCM2835 (4) @ 1.800GHz 
Memory: 3295MiB / 7812MiB 

PHP 8.2.5 (cli) (built: Apr 14 2023 04:39:46) (NTS)

out of the box install with apt install cacti
running apache

web interface works, but poller not:

$ php /usr/share/cacti/site/poller.php
PHP Warning:  Trying to access array offset on value of type null in /usr/share/cacti/site/lib/functions.php on line 4574
PHP Fatal error:  Uncaught TypeError: round(): Argument #1 ($num) must be of type int|float, string given in /usr/share/cacti/site/poller.php:333
Stack trace:
#0 /usr/share/cacti/site/poller.php(333): round()
#1 {main}
  thrown in /usr/share/cacti/site/poller.php on line 333

If this is already recorded as issue and/or fixed, i searched a lot and didn't find anything, sorry for this!

**Comment by TheWitness** [comment]
What Cacti version?

**Comment by ufehembach** [comment]
Version 1.2.16

**Comment by TheWitness** [comment]
Either back off PHP versions to something like PHP7.4, or upgrade to Cacti 1.2.24.

**Comment by TheWitness** [comment]
PHP 8 was a big change, and lot's of things broke not just in Cacti, but around the world.  It was a good change and long overdue.

**Comment by ufehembach** [comment]
I know, may systems stopped working. i don't get librenms running oob so i tried to switch to cacti. doesn"t work either.
I'm too tired to work around. if apt install doesn't work i don't use it.
got 4 boxes on different sites to manage.
will wait for next update.
thanks. you can close this issue.

**Comment by TheWitness** [comment]
When you are running from Apt, upgrading is a bit more complicated.  When upgrading from source, it's quite simple.  Best ask the Debian guys to get you a link to a more recent repo.  From this chart, it looks like they have it built already:

https://tracker.debian.org/teams/cacti-team/

**Comment by netniV** [comment]
If you are after any of the latest cacti versions, you need the backports repository from Paul I think.  Personally, I prefer just to use git for updates instead.  So much easier as upgrading is a one liner (`git checkout release/1.2.24` for example).  TheWitness prefers many source copying which is documented.

**Comment by ufehembach** [comment]
Hi, ok, thanks for the input. I tried this git thing but I can not get along with the request. I also tried gh stuff as given on github but failed with the auth. So i downloaded the develop.zip, extracted it and started the installation. 
I did felt 20 changes to my mysql and the installation runs very smooth, no warnings anymore a lot of confirmations and so on.
And when I'm at the end of the installation it starts again. And if I restart my server and got to http://server/cacti  installation starts again. So what I can say, installation works fine.
